### DISCUSSION TITLE
Always attach playstyle to osu! scores before submission

### DISCUSSION THREAD

**Discussion by Awesome-Aasim** [ppy/osu#30840] [Ideas]
This is already done for touchscreen plays, largely because touchscreen plays compute performance differently.

I think it would be helpful to always attach playstyle because different modes have different advantages. While touchscreen reduces difficulty in hitting jumps, digitizer tablets reduce difficulty with aim, and keyboard reduces difficulty for streams, with mouse being one of the hardest. Increasing difficulty in touch would probably require more charts with simultaneous objects like circles and sliders on the same point in the timeline.

I also think filtering scores by playstyle on leaderboards should not require an osu!supporter tag, because playstyles are not mods; they are ways for players to enjoy the game. This is especially going to be problematic when osu! is released on the App Store and Play Store on iOS and Android; most will not have a digitizer to plug into their phone or tablet, they will just play with only the touchscreen.

**Comment by WalmartWhiteCat** [comment]
long story short touchscreen easy, other things not easy. Done. Read on if you want a way-too-long comment because it’s late and I can’t sleep.



Sorry but this idea is just kinda bad. The reason touchscreen is displayed is because it objectively makes maps easier across the board with little downside. You can still use a keyboard while doing touchscreen and mobile osu! has support for using the side of the screen as a keypad, not to mention you can get and use a stylus on mobile to remove the downside of having to use your finger to aim. This basically means you get all the upside of near-instant aim, being able to see your hand/stylus and exactly where it’s going and none of the downside of actually using a touch device (which would be the inability to easily see patterns under one’s finger and alternating being more difficult).

And to touch (pun intended) on your other points, I personally think they’re all wrong. Pretty much every single play style is no more than preference (yes even touchscreen, but that invalidates none of my points). Sure, one might seem harder to you than another, but consider the fact that the people using them are used to them. Not to mention the fact that every style has its ups and downs (again, even touchscreen, but to a lesser extent).

Mouse has the pro of steady aim with the con of drift.
Tablet has the pro of absolute positioning with the con of an ever-changing/unstable grip.
I am confident that any other play style will follow this pattern, even the most gimmicky of the bunch (though each will be to a greater or lesser extent). But one fact holds true no matter what. If you can get used to it, you can become a pro with it.

So, to refresh, touchscreen makes virtually every single map easier, with the ability to act as a tablet and keyboard while still allowing you to effectively teleport your cursor and hit even the most impossible of jumps (as long as the objects are physically on-screen). No other legitimate play style even comes close to this level of aim and versatility. If you play on tablet you have issues with grip that are extremely difficult to eliminate. If you play on mouse you have issues with drift that are extremely difficult to eliminate. If you play on touchscreen you get to teleport your cursor while everything else remains almost the same.

Sorry for harping on but it’s just a thing that should never be understated. Touchscreen as a whole opens more paths for insane cheese mechanics than anything else while keeping other elements of gameplay virtually the same or at least keeping them very easy to adapt to despite being a whole new play style.

**Reply by Awesome-Aasim** [reply]
I do not disagree that touchscreen makes certain plays easier. In fact, certain kinds of gameplay are _only_ possible on touchscreen, like hitting two simultaneous notes numbered the same on the same timeline tick as in osu!stream; two notes on the same timeline tick doesn't happen to ranked charts at this time because of the ranking criteria. Also certain jump plays like that on Tag4 charts that were demoted from "Ranked" to "Loved". But even if nothing about the score calculation took into account the playstyle for modes other than touchscreen, it is still useful to filter by playstyle. For example, seeing touchscreen only vs. touchscreen + mouse/keyboard vs. mouse only vs. mouse + keyboard/second mouse vs. tablet vs. tablet + keyboard, etc.

**Reply by WalmartWhiteCat** [reply]
Consider the fact that touchscreen can make the hardest ranked maps in the world relatively simple for people with touchscreen. It's not just the TAG4 maps, I implied that as an example. All maps with any level of jump aim difficulty become nearly trivial in the face of touchscreen. And, as mentioned before, they still get to use keyboards and keyboard-like mechanics to aid in alternating. I mean, even the first 10 star FC was on touchscreen, with a non-touchscreen FC coming into existence far later.

It's a play style that objectively alters the game too much for it to not be listed on scores, as a TD play at the upper echelon is simply far less impressive than a play of the same caliber on a mouse/tablet play style (But don't get me wrong, touchscreen players are still good at the game, just not in the same way as other players). There's just no way around it. With the nature of the style it's practically impossible to compare it to anything else.

**Comment by cl8n** [comment]
like WalmartWhiteCat i just don't see this as necessary and the premise seems too opinionated as well.

> I also think filtering scores by playstyle on leaderboards should not require an osu!supporter tag, because playstyles are not mods; they are ways for players to enjoy the game. This is especially going to be problematic when osu! is released on the App Store and Play Store on iOS and Android; most will not have a digitizer to plug into their phone or tablet, they will just play with only the touchscreen.

*maybe* this part could be worth considering but I think it'd have to wait until we see what affect mobile releases have on leaderboards. currently we can only guess
